src/celeritas/optical/ScintillationParams.cc

```diff
@@ -36,11 +39,67 @@ ScintillationParams::from_import(ImportData const& data)
         return nullptr;
     }
 
+    auto const& num_optmats = data.optical.size();
+
     Input input;
-    for (auto const& mat : data.optical)
+    input.resolution_scale.resize(num_optmats);
+    if (data.optical_params.scintillation_by_particle)
+    {
+        // Collect ScintillationParticleIds
+        input.pid_to_scintpid.resize(data.particles.size());
+        ScintillationParticleId scintpid;
+        for (auto const& [matid, iom] : data.optical)
+        {
+            auto const& iomsp = iom.scintillation.particles;
+            for (auto const& [pdg, ipss] : iomsp)
+            {
+                auto const pid = particle_params->find(PDGNumber{pdg});
+                // Add new ScintillationParticleId
+                input.pid_to_scintpid[pid.get()] = scintpid++;
```

Comment:
```suggestion
                if (auto const pid = particle_params->find(PDGNumber{pdg}))
                {
                    // Add new ScintillationParticleId
                    input.pid_to_scintpid[pid.get()] = scintpid++;
                }
```
Since electrons are the only particles of the optical particle types supported in Celeritas, only the electron particle-dependent spectrum data will ever be stored/used, right?

Reply:
That is a good point. Scratch my other comment about storing data from particles we do not have in our `ScintillationParams`, because it is wrong. This happens in the `GeantImporter`, but not in the `ScintillationParams`, since we depend on `ParticleParams` to begin with...